packages/synapse-react-client/src/components/AccessRequirementList/AccessRequirementList.tsx

```diff
@@ -217,6 +220,21 @@ export default function AccessRequirementList(
     fetchedRequirementsForEntity ??
     fetchedRequirementsForTeam
 
+  // SWC-7218: Group by access requirement type
+  const groupedAccessRequirementsByType = useMemo(() => {
+    if (accessRequirements) {
+      return accessRequirements.reduce((acc, item) => {
+        // init if we have not seen this concrete type before
+        if (!acc[item.concreteType]) {
+          acc[item.concreteType] = []
+        }
+        acc[item.concreteType].push(item)
+        return acc
+      }, {} as Record<AccessRequirement['concreteType'], AccessRequirement[]>)
+    }
+    return undefined
+  }, [accessRequirements])
```

Comment:
Can you use lodash `groupBy`?

```suggestion
  const groupedAccessRequirementsByType: Record<AccessRequirement['concreteType'], AccessRequirement[]> = useMemo(() => {
    if (accessRequirements) {
      return groupBy(accessRequirements, 'concreteType')
    }
    return undefined
  }, [accessRequirements])
```